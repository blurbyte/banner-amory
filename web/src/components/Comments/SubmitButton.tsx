import * as React from 'react';

import { PaperPlane } from '../Icons'
import Button from '../Button';

class SubmitButton extends React.Component {
  render() {
    return (
      <Button type="submit" icon={PaperPlane} aria-label="Comment">Comment</Button>
    );
  }
}

export default SubmitButton;
