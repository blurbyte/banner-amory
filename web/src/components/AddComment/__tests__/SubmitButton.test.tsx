import * as React from 'react';
import * as renderer from 'react-test-renderer';
import 'jest-styled-components';

import SubmitButton from '../SubmitButton';

test('renders correctly', () => {
  const component = renderer.create(<SubmitButton />);
  expect(component.toJSON()).toMatchSnapshot();
});
