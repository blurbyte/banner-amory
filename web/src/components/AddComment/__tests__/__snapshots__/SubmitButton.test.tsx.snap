// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  width: 100%;
  height: 5rem;
  width: 15rem;
  background: rgba(61,38,218,0.1);
  padding: 1.4rem 1rem 0 1rem;
  border: 0;
  outline: 0;
  border-top: 4px solid transparent;
  cursor: pointer;
  font-size: 1.8rem;
  font-weight: 700;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: all 0.15s linear;
  transition: all 0.15s linear;
}

.c0:focus {
  border-top: 4px solid #3d26da;
}

.c1 {
  position: absolute;
  bottom: -1rem;
  left: -2rem;
}

<button
  className="c0"
  type="submit"
>
  <div
    className="c1"
  >
    <svg
      height="38"
      viewBox="0 0 46 38"
      width="46"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M5.972 12.497s-2.845.837-3.746 1.236c-.094.042-2.135.473-2.226.516 1.209.849 2.82 1.155 3.539 2.49.081-.019 1.374.582 1.457.564l2.757 1.806c.48.055 1.027.517 1.2.786l5.043 3.303-.404 13.809 7.745-8.226 3.121 2.026c1.341.635 2.668 1.592 3.944 2.226 1.798.893 2.08 2.061 3.847 3.014.377.203 1.634.812 3.039 1.954.156-1.882.482-3.276.791-4.272C38.688 25.334 46 0 46 0L5.972 12.497z"
        fill="#e2abff"
      />
      <path
        d="M18.648 20.386l-.055 10.065-3.666 3.705.066-10.971 3.655-2.799zM6.001 16.479l8.478 5.553 3.901-2.395-11.816-6.29s-1.751.464-3.025.876c.676.64 2.333 2.117 2.462 2.256zM43.463 2.067l.617-.432L7.228 13.14l11.607 6.179L43.463 2.067zm-1.71 1.829L19.565 19.731l-.342 9.037 22.53-24.872zm2.041-1.463L25.552 22.896l10.473 5.576 8.248-26.577-.479.538zM29.516 28.937l-.525 2.849s3.99 2.325 5.006 3.222l.003-.009 1.876-6.046-10.663-5.677-3.25 3.948 5.338 3.465 2.215-1.752z"
        fill="#3d26da"
      />
    </svg>
  </div>
  Comment
</button>
`;
